crates/transaction-pool/src/pool/parked.rs

```diff
@@ -95,6 +95,18 @@ impl<T: ParkedOrd> ParkedPool<T> {
             .collect()
     }
 
+    /// Get transactions, with size, by sender
+    pub(crate) fn get_txs_by_sender_with_size(
+        &self,
+        sender: SenderId,
+    ) -> Vec<(TransactionId, usize)> {
+        self.by_id
+            .range((sender.start_bound(), Unbounded))
+            .take_while(move |(other, _)| sender == other.sender)
+            .map(|(tx_id, tx)| (*tx_id, tx.transaction.size()))
+            .collect()
+    }
```

Comment:
do we need this if we don't use the size in the truncate loop?

Reply:
I think we can solve this even with out allocating a vec and instead find the highest nonce of the sender and remove that

Reply:
Ah, we don't, artifact of an earlier fix

Reply:
actually maybe allocating is better than several lookups